src/overlays/actors/ovl_En_Kakasi/z_en_kakasi.c

```diff
@@ -46,80 +77,1175 @@ const ActorInit En_Kakasi_InitVars = {
+    } else if (ActorCutscene_GetCanPlayNext(this->actorCutscenes[cutsceneIndex]) == 0) {
+        ActorCutscene_SetIntentToPlay(this->actorCutscenes[cutsceneIndex]);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_80971064.s")
+    } else {
+        ActorCutscene_StartAndSetUnkLinkFields(this->actorCutscenes[cutsceneIndex], &this->actor);
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_AKINDONUTS_HIDE);
+        this->actor.draw = EnKakasi_Draw;
+        this->unkState196 = 6;
+        this->actionFunc = EnKakasi_RisingOutOfGround;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_80971430.s")
+void EnKakasi_RisingOutOfGround(EnKakasi* this, GlobalContext* globalCtx) {
+    this->actor.shape.rot.y += 0x3000;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_80971440.s")
+    if (this->animeIndex != ENKAKASI_ANIME_SIDEWAYS_SHAKING) {
+        EnKakasi_SetAnimation(this, ENKAKASI_ANIME_SIDEWAYS_SHAKING);
+    }
+    if (this->actor.shape.yOffset < -10.0f) {
+        if ((globalCtx->gameplayFrames & 7) == 0) {
+            func_800BBDAC(globalCtx, &this->actor, &this->actor.world.pos, this->actor.shape.shadowScale - 20.0f, 10,
+                          8.0f, 500, 10, 1);
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_AKINDONUTS_HIDE);
+        }
+        Math_ApproachF(&this->actor.shape.yOffset, 0.0f, 0.5f, 200.0f);
+    } else {
+        EnKakasi_SetupIdleRisen(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_809714BC.s")
+void EnKakasi_SetupIdleRisen(EnKakasi* this) {
+    // "Yowza! Oh, yeah!, We meet again, baby! What great playing, as usual!"
+    this->actor.textId = 0x164D;
+    this->unkState196 = 7;
+    this->actionFunc = EnKakasi_IdleRisen;
+    this->actor.shape.yOffset = 0.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_80971794.s")
+void EnKakasi_IdleRisen(EnKakasi* this, GlobalContext* globalCtx) {
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, 1000, 0);
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        this->actionFunc = EnKakasi_RisenDialogue;
+    } else {
+        func_800B8614(&this->actor, globalCtx, 70.0f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_809717D0.s")
+void EnKakasi_RisenDialogue(EnKakasi* this, GlobalContext* globalCtx) {
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, 1000, 0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_8097185C.s")
+    if (func_80152498(&globalCtx->msgCtx) == 5 && func_80147624(globalCtx) != 0) {
+        func_801477B4(globalCtx);
+        EnKakasi_SetupIdleRisen(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_8097193C.s")
+void EnKakasi_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnKakasi* this = (EnKakasi*)thisx;
```

Comment:
```suggestion
    EnKakasi* this = THIS;
```

Reply:
Oh I thought we were getting rid of THIS, I already removed the macro...

Reply:
We are going to do that eventually, but not yet :sweat_smile: 
It would probably be a "all at the same time" change.